core/common/src/main/java/alluxio/worker/block/io/BlockReader.java

```diff
@@ -82,4 +82,17 @@ public BlockReader() {
    * @return an informational string of the location the reader is reading from
    */
   public abstract String getLocation();
+
+  /**
+   * Abort the reader. This method should be called when an exception occurs during execution
+   * or when the operation is cancelled by the user.
+   */
+  public void abort() throws IOException {
+  }
+
+  /**
+   * Commit the reader in case there is any uncommitted block cached from UFS.
+   */
+  public void commit() throws IOException {
```

Comment:
I don't see why need a separate `commit()`, let's discuss offline

Reply:
Yes we can combine these two functions but I feel it's more clear to separate the responsibility of close resources & commit block.